src/raw/mod.rs

```diff
@@ -2152,14 +2238,12 @@ impl<A: Allocator + Clone> RawTableInner<A> {
         debug_assert!(self.items <= capacity);
 
         // Allocate and initialize the new table.
-        let mut new_table = RawTableInner::fallible_with_capacity(
+        let new_table = RawTableInner::fallible_with_capacity(
             self.alloc.clone(),
             table_layout,
             capacity,
             fallibility,
         )?;
-        new_table.growth_left -= self.items;
-        new_table.items = self.items;
```

Comment:
Why are these moved?

Reply:
They are not needed here because:
1. This allows you to make the function safe.
2. Adds more consistency in functions, allowing less overthinking and remembering that you changed the number of elements before actually adding them, which is not a good idea in my opinion and only confuses. For example, in the `clone_from_impl` method, we first add items, and only then change `self.table.items` and `self.table.growth_left` fields.